haystack/document_stores/search_engine.py

```diff
@@ -105,10 +105,12 @@ def __init__(
             raise DocumentStoreError(
                 f"Invalid value {similarity} for similarity, choose between 'cosine', 'l2' and 'dot_product'"
             )
-        if index_type in ["flat", "hnsw"]:
+        if index_type in ["flat", "hnsw", "ivf", "ivf_pq"]:
```

Comment:
The allowed values could be extracted into a class variable. That way `OpenSearchDocumentStore` could extend them, so we don't need to falsely allow them for `ElasticsearchDocumentStore`.

Reply:
I removed `index_type` entirely from `ElasticsearchDocumentStore` as we don't use this parameter with Elasticsearch.